### PR TITLE
fix: update assignment policy logic

### PR DESCRIPTION
**Pull request by ciduarte**
[AA-836](https://openedx.atlassian.net/browse/AA-836)

### CODE REVIEW

File: src/course-home/data/api.js [hunk 1]
```diff
@@ -35,22 +35,27 @@ function normalizeAssignmentPolicies(assignmentPolicies, sectionScores) {
...
       const {
         assignmentType,
         numPointsEarned,
         numPointsPossible,
       } = subsection;
-      let {
-        numAssignmentsCreated,
-      } = gradeByAssignmentType[assignmentType];
 
-      if (!numPointsPossible) {
```
Comment (ciduarte) [inline src/course-home/data/api.js:50]: moved this as a nit of mine

File: src/course-home/data/api.js [hunk 2]
```diff
@@ -35,22 +35,27 @@ function normalizeAssignmentPolicies(assignmentPolicies, sectionScores) {
...
+      // If a subsection's assignment type does not match an assignment policy in Studio,
+      // we won't be able to include it in this accumulation of grades by assignment type.
+      // This may happen if a course author has removed/renamed an assignment policy in Studio and
+      // neglected to update the subsection's of that assignment type
```
Comment (ciduarte) [inline src/course-home/data/api.js:50]: Not sure if this comment is overkill. This behavior wasn't obvious to me so it felt worth documenting

File: src/course-home/data/api.js [hunk 3]
```diff
@@ -35,22 +35,27 @@ function normalizeAssignmentPolicies(assignmentPolicies, sectionScores) {
...
+      if (!gradeByAssignmentType[assignmentType]) {
```
Comment (MatthewPiatetsky) [inline src/course-home/data/api.js:51]: couldn't you do the below statement and then check if (!numAssignmentsCreated) to avoid using the same code twice?

**Review by MatthewPiatetsky — APPROVED** [review]

File: src/course-home/data/api.js (hunk 3 above)
Comment (ciduarte) [reply src/course-home/data/api.js:51]: The bug that was happening in prod was due to trying to retrieve `numAssignmentsCreated` from null, so we need to check if `gradeByAssignmentType[assignmentType]` exists first.

**Pull request merged**